api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/services/cached/CachedApiDocService.java

```diff
@@ -48,34 +49,63 @@ public CachedApiDocService(APIDocRetrievalService apiDocRetrievalService, Transf
     }
 
     /**
-     * Update the api docs for this service
+     * Fetches API documentation for a given service and version.
      *
-     * @param serviceId  service identifier
-     * @param apiVersion the version of the API
-     * @return api doc info for the requested service id
+     * @param serviceId    service identifier
+     * @param cacheKeySuffix version or default API key
+     * @param retrievalLogic supplier providing the API documentation retrieval logic
+     * @return the API documentation
      */
-    public String getApiDocForService(final String serviceId, final String apiVersion) {
-        // First try to fetch apiDoc from the DS
+    private String fetchApiDoc(
+        final String serviceId,
+        final String cacheKeySuffix,
+        final Supplier<ApiDocInfo> retrievalLogic
+    ) {
+        ApiDocCacheKey cacheKey = new ApiDocCacheKey(serviceId, cacheKeySuffix);
+        String errorMessage = "";
+        Exception exception = null;
+
+        // Try to fetch API doc from the retrieval logic
         try {
-            ApiDocInfo apiDocInfo = apiDocRetrievalService.retrieveApiDoc(serviceId, apiVersion);
+            ApiDocInfo apiDocInfo = retrievalLogic.get();
             if (apiDocInfo != null && apiDocInfo.getApiDocContent() != null) {
```

Comment:
should it have a debug message if no exception is thrown but there was no doc?

Reply:
It makes sense, thanks